runtime/parachains/src/ump.rs

```diff
@@ -537,6 +544,7 @@ impl<T: Config> Pallet<T> {
 				match T::UmpSink::process_upward_message(dispatchee, upward_message, max_weight) {
 					Ok(used) => {
 						weight_used += used;
+						messages_processed += 1;
```

Comment:
wrong place - since the `Err` branch will also use (PoV) weight for which we are not accounting. Must be at the root.